src/hints/execution.rs

```diff
@@ -326,3 +361,409 @@ pub fn exit_call(
     execution_helper.exit_call();
     Ok(())
 }
+
+pub const CONTRACT_ADDRESS: &str = indoc! {r#"
+    from starkware.starknet.business_logic.transaction.deprecated_objects import (
+        InternalL1Handler,
+    )
+    ids.contract_address = (
+        tx.contract_address if isinstance(tx, InternalL1Handler) else tx.sender_address
+    )"#
+};
+
+pub fn contract_address(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let tx = exec_scopes.get::<InternalTransaction>("tx")?;
+    let contract_address = if tx.r#type == "L1_HANDLER" {
```

Comment:
Out of curiosity, what is this `tx.r#type` syntax?

Reply:
Not sure. `r#type` is an identifier. Not written by me, maybe just a typo?